src/anomalib/deploy/inferencers/torch_inferencer.py

```diff
@@ -206,10 +206,15 @@ def pre_process(self, image: np.ndarray) -> torch.Tensor:
         Returns:
             Tensor: pre-processed image.
         """
-        if len(image) == 3:
-            image = image.unsqueeze(0)
+        if len (image.shape) == 3:
+            transposed_image = image.transpose(2,0,1).copy()
+        else:
+            transposed_image = image.copy()
 
-        return image.to(self.device)
+        if len(transposed_image) == 3:
+            return torch.from_numpy(transposed_image).float().to(self.device).unsqueeze(0)
+        else : 
+            return torch.from_numpy(image).float().to(self.device).unsqueeze(0)
```

Comment:
Again, this is to pass the code-quality checks.
```suggestion
        if len(transposed_image) == 3:
            return torch.from_numpy(transposed_image).float().to(self.device).unsqueeze(0)

        return torch.from_numpy(image).float().to(self.device).unsqueeze(0)
```